File: example/ios/Runner/SamplePlatformView.swift
//
//  SamplePlatformView.swift
//  Runner
//
//  Created by 砂賀開晴 on 2022/08/27.
//

import Foundation
import UIKit
import extended_platform_view

class SamplePlatformView: ExtendedPlatformView {
    override func initialize(params: CreationParams) -> UIView {
        let label = UILabel(frame: params.frame)
        label.text = (params.args as! String)
        
        methodChannel.setMethodCallHandler({ call, result in
            switch call.method {
            case "set_text":
                label.text = (call.arguments as! String)
                result(nil)
            default:
                result(FlutterMethodNotImplemented)
            }
        })
        
        return label
    }
}

File: example/android/app/src/main/kotlin/work/kscafe/extended_platform_view_example/SamplePlatformView.kt
package work.kscafe.extended_platform_view_example

import android.view.View
import android.widget.TextView
import work.kscafe.extended_platform_view.CreationParams
import work.kscafe.extended_platform_view.ExtendedPlatformView

class SamplePlatformView: ExtendedPlatformView() {
    override fun initialize(params: CreationParams): View {
        val textView = TextView(context).apply {
            text = params.args as String
        }

        // ExtendedPlatformView generates MethodChannel automatically
        methodChannel.setMethodCallHandler { call, result ->
            when(call.method) {
                "set_text" -> {
                    textView.text = call.arguments as String
                    result.success(null)
                }
                else -> result.notImplemented()
            }
        }

        return textView
    }
}
